subsys/bluetooth/mesh/adv.c

```diff
@@ -47,124 +46,144 @@ static K_FIFO_DEFINE(bt_mesh_adv_queue);
 static K_FIFO_DEFINE(bt_mesh_relay_queue);
 static K_FIFO_DEFINE(bt_mesh_friend_queue);
 
-void bt_mesh_adv_send_start(uint16_t duration, int err, struct bt_mesh_adv *adv)
+K_MEM_SLAB_DEFINE(adv_buf_pool, sizeof(struct bt_mesh_adv),
+		  CONFIG_BT_MESH_ADV_BUF_COUNT, __alignof__(struct bt_mesh_adv));
```

Comment:
It is better to not rely on some implicit assumptions. Yes, at the moment pointers are aligned but after some changes and  PRs might not be aligned. Anycase since slab allocator definition API takes care about alignment it is not the issue.